### PR TITLE
Update predict pipeline

### PR DESCRIPTION
**Pull request by Mehrad0711**
Add option to do prediction on pretrained models.
Related PR in genienlp: https://github.com/stanford-oval/genienlp/pull/107

### CODE REVIEW

**Comment by s-jse** [comment]
Why do we need the `prediction-small` pipeline? The cost of having 4 GPUs in parallel is almost the same as having one GPU machine for longer.
Also, the number of GPUs can be an input parameter, like what we have for training pipelines.

**Comment by Mehrad0711** [comment]
Yeah but I noticed for predictions on small datasets, the actual generation time is small compared to total pod time that it would be cheaper to use g4dn.4xlarge. In fact, 2xlarge or even xlarge should be sufficient.

**Review by gcampax — COMMENTED** [review]
I'm not sure i understand this PR, it creates new small pipelines, but the small pipelines are still g4dn.4xlarge?

**Comment by s-jse** [comment]
> 
> 
> I'm not sure i understand this PR, it creates new small pipelines, but the small pipelines are still g4dn.4xlarge?

It has one GPU instead of 4.

**Review by s-jse — CHANGES_REQUESTED** [review]
Instead of a new pipeline, number of GPUs can be an input parameter. Similar to training pipelines (https://github.com/stanford-oval/genie-k8s/blob/master/pipelines/training.py#L107), but instead of a constant in the code, surfaced to the kubeflow experiment UI as well.

**Comment by Mehrad0711** [comment]
num_gpus is a runtime parameter, it won't be available at compile-time. So you can't pass it as a Pipeline parameter and parse it in python. Additionally, the node selector dictates number of gpus so I can't reuse `predict` pipeline that uses `g4dn.12xlarge` with only 1 gpu.

File: pipelines/training.py [hunk 1]
```diff
@@ -1412,33 +1461,136 @@ def eval_only_pipeline(
...
         s3_input_datadir=s3_input_datadir,
+        model_type=model_type,
         dataset_subfolder=dataset_subfolder,
         val_batch_size=val_batch_size,
         additional_args=additional_args,
         genienlp_version=genienlp_version
         )
 
+@dsl.pipeline(
+    name='Predict',
```
Comment (s-jse) [inline pipelines/training.py]: Use a unique name for pipelines.

**Review by s-jse — APPROVED** [review]
Please add the new pipelines to `upload_all_pipelines.sh` as well.

**Comment by Mehrad0711** [comment]
> Please add the new pipelines to `upload_all_pipelines.sh` as well.

I'm sending a followup PR which has some updates including this (I needed to add bootleg pipelines too).

**Comment by Mehrad0711** [comment]
Will update predict pipelines with this merge though.

**Pull request merged**